pkg/segment/structs/evaluationstructs.go

```diff
@@ -712,6 +712,95 @@ func (self *BoolExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure) (b
 	}
 }
 
+// This evaluation is specific for inputlookup command.
+// Only =, != , <, >, <=, >= operators are supported for strings and numbers.
+// Strings and numbers are compared as strings.
+// Strings are compared lexicographically and are case-insensitive.
+// Wildcards can be used to match a string with a pattern.
+func (self *BoolExpr) EvaluateForInputLookup(fieldToValue map[string]utils.CValueEnclosure) (bool, error) {
+	if self.IsTerminal {
+		leftStr, errLeftStr := self.LeftValue.EvaluateToString(fieldToValue)
+		rightStr, errRightStr := self.RightValue.EvaluateToString(fieldToValue)
+		leftFloat, errLeftFloat := self.LeftValue.EvaluateToFloat(fieldToValue)
+		rightFloat, errRightFloat := self.RightValue.EvaluateToFloat(fieldToValue)
+
+		if errLeftFloat == nil && errRightFloat == nil {
+			switch self.ValueOp {
+			case "=":
+				return leftFloat == rightFloat, nil
+			case "!=":
+				return leftFloat != rightFloat, nil
+			case "<":
+				return leftFloat < rightFloat, nil
+			case ">":
+				return leftFloat > rightFloat, nil
+			case "<=":
+				return leftFloat <= rightFloat, nil
+			case ">=":
+				return leftFloat >= rightFloat, nil
+			default:
+				return false, fmt.Errorf("BoolExpr.EvaluateForInputLookup: invalid ValueOp %v for floats", self.ValueOp)
+			}
+		} else if errLeftStr == nil && errRightStr == nil {
+			leftStr = strings.ToLower(leftStr)
+			rightStr = strings.ToLower(rightStr)
+			match, err := filepath.Match(rightStr, leftStr)
```

Comment:
`filepath.Match` does more than just allowing `*`. Do you really want `filepath.Match`, or do you just want to allow `*` as "match anything, any number of characters"? If you just want `*`, you might want to use https://github.com/siglens/siglens/blob/develop/pkg/common/dtypeutils/dtypeutils.go#L599-L616

Reply:
addressed in the latest commit.